From e99379db09d83ab2c298087112aa2757c8bb7108 Mon Sep 17 00:00:00 2001
From: Min He <min.he@intel.com>
Date: Wed, 31 Jan 2018 16:26:16 +0800
Subject: [PATCH 0520/1240] drm/i915/gvt: fixed some compilation issues

This patch fixed some compilation issues in display.c and firmware.c,
which were introduced by previous patches.

Signed-off-by: Min He <min.he@intel.com>
(cherry picked from commit 3acf01fd06bf02f63338f7e14e3c97f908a8511f)

Change-Id: I4675de420e38b8b1f8251bc02412b98efcc11d9b
---
 drivers/gpu/drm/i915/gvt/display.c  | 24 +++++++++++++-----------
 drivers/gpu/drm/i915/gvt/firmware.c |  5 +----
 2 files changed, 14 insertions(+), 15 deletions(-)

diff --git a/drivers/gpu/drm/i915/gvt/display.c b/drivers/gpu/drm/i915/gvt/display.c
index c6c15b2..71279fa 100644
--- a/drivers/gpu/drm/i915/gvt/display.c
+++ b/drivers/gpu/drm/i915/gvt/display.c
@@ -312,19 +312,21 @@ static int setup_virtual_monitor(struct intel_vgpu *vgpu, int port_num,
 	if (!port->edid)
 		return -ENOMEM;
 
-		port->dpcd = kzalloc(sizeof(*(port->dpcd)), GFP_KERNEL);
-		if (!port->dpcd) {
-			kfree(port->edid);
-			return -ENOMEM;
-		}
+	port->dpcd = kzalloc(sizeof(*(port->dpcd)), GFP_KERNEL);
+	if (!port->dpcd) {
+		kfree(port->edid);
+		return -ENOMEM;
+	}
 
-		if (edid)
-			memcpy(port->edid->edid_block, edid, EDID_SIZE * valid_extensions);
-		else
-			memcpy(port->edid->edid_block, virtual_dp_monitor_edid[resolution],
-					EDID_SIZE);
+	if (edid)
+		memcpy(port->edid->edid_block, edid,
+				EDID_SIZE * valid_extensions);
+	else
+		memcpy(port->edid->edid_block,
+				virtual_dp_monitor_edid[resolution],
+				EDID_SIZE);
 
-		port->edid->data_valid = true;
+	port->edid->data_valid = true;
 
 	if (is_dp) {
 		memcpy(port->dpcd->data, dpcd_fix_data, DPCD_HEADER_SIZE);
diff --git a/drivers/gpu/drm/i915/gvt/firmware.c b/drivers/gpu/drm/i915/gvt/firmware.c
index 9f8a0b6..3204c5e 100644
--- a/drivers/gpu/drm/i915/gvt/firmware.c
+++ b/drivers/gpu/drm/i915/gvt/firmware.c
@@ -221,16 +221,13 @@ int intel_gvt_load_firmware(struct intel_gvt *gvt)
 	int ret;
 
 	mem = kmalloc(info->cfg_space_size, GFP_KERNEL);
-	if (!mem) {
-		kfree(path);
+	if (!mem)
 		return -ENOMEM;
-	}
 
 	firmware->cfg_space = mem;
 
 	mem = kmalloc(info->mmio_size, GFP_KERNEL);
 	if (!mem) {
-		kfree(path);
 		kfree(firmware->cfg_space);
 		return -ENOMEM;
 	}
-- 
2.7.4

